resources/js/provider.js

```diff
@@ -152,6 +159,25 @@
     EthereumProvider.prototype.status = new StatusAPI();
     EthereumProvider.prototype.isConnected = function () { return true; };
 
+    EthereumProvider.prototype._events = {};
+
+    EthereumProvider.prototype.on = function(name, listener) {
+        if (!this._events[name]) {
+          this._events[name] = [];
+        }
+        this._events[name].push(listener);
+    }
+
+    EthereumProvider.prototype.removeListener = function (name, listenerToRemove) {
+        if (!this._events[name]) throw new Error(`event "${name}" does not exist`);
+        const filterListeners = (listener) => listener !== listenerToRemove;
+        this._events[name] = this._events[name].filter(filterListeners);
+    }
+
+    EthereumProvider.prototype.emit = function (name, data) {
+        if (!this._events[name]) throw new Error(`event "${name}" does not exist`);
```

Comment:
hm not sure we need to throw event here, what if dapp is not subscribed for accountsChanged?

Reply:
actually you are right